redis_dict.py

```diff
@@ -214,6 +214,7 @@ def __init__(self,
                  namespace: str = 'main',
                  expire: Union[int, timedelta, None] = None,
                  preserve_expiration: Optional[bool] = False,
+                 redis: Optional[StrictRedis[Any]] = None,
```

Comment:
The Redis class isn't generic, but it is generic within types-redis.
This causes to types-redis to mask redis at type checking time, but not at runtime.
Python 3.11 apparently does its own type checks at runtime, but I don't know why it doesn't do that within GitHub actions tho...
Never happened to me before, but it kind of makes sense since generics got their own syntax with python 3.12.

Also note that redis 5.0 comes with its own py.typed marker. So types-redis will not be needed then.